src/net_processing.cpp

```diff
@@ -1720,7 +1720,7 @@ void PeerManagerImpl::InitializeNode(const CNode& node, ServiceFlags our_service
     NodeId nodeid = node.GetId();
     {
         LOCK(cs_main); // For m_node_states
-        m_node_states.emplace_hint(m_node_states.end(), std::piecewise_construct, std::forward_as_tuple(nodeid), std::forward_as_tuple(node.IsInboundConn()));
+        m_node_states.try_emplace(m_node_states.end(), nodeid);
```

Comment:
should we assert the result of this code instead of dropping it on the floor?

Reply:
We were not checking it before, nor are we doing so later when emplacing it in `m_peer_map` either, so I'd lean towards no for consistency, but I don't have a strong opinion about it